### PR TITLE
[ll] Metal Backend: Support multiple windowing libs

### PR DESCRIPTION
**Pull request by davll**
Metal backend in `ll` branch needed winit object to create surface. However, it was inconvenient for users with sdl2, glew3 windowing library instead of winit library. Thus I made a trait NativeWindow to bridge window objects and the metal Instance in gfx-rs.

### CODE REVIEW

**Comment by kvark** [comment]
I feel that we better have `winit` as default (either built-in or as a default feature), with extra windowing stuff like sdl/glfw coming as extra features. `winit` is available everywhere, and it's rusty, means that it's a natural fit to work with gfx-rs. We should encourage it, not gate it away.

**Comment by davll** [comment]
@msiglreith Thanks for approval :D
However, I discovered `NativeWindow trait` approach might be problematic for Vulkan backend and other backends, since it has to detect WinAPI, Xlib, XCB, and Wayland. Should `NativeWindow trait` be specific for gfx_backend_* or defined in gfx_backend_common (the shared lib for gfx backends)?

@kvark Is it OK to make `winit` feature default?

**Comment by davll** [comment]
I've rebased the commits :)

File: src/backend/metal/src/lib.rs [hunk 1]
```diff
@@ -61,11 +77,9 @@ impl Instance {
         Instance {}
     }
 
-    pub fn create_surface(&self, window: &winit::Window) -> Surface {
+    pub fn create_surface(&self, window: &NativeWindow) -> Surface {
```
Comment (kvark) [inline src/backend/metal/src/lib.rs]: I don't think having a trait here is really necessary. We don't need to be generic here. Instead, we could have:
```rust
pub fn create_surface_from_nsview(&self, ns_view: cocoa::base::id) -> Surface {...}
#[cfg(feature="winit")]
pub fn create_surface(&self, window: &winit::Window) -> Surface {
    use winit::os::macos::WindowExt;
    self.create_surface_from_nsview(WindowExt::get_nsview(window))
}
```

**Comment by davll** [comment]
@kvark Nice point! Thanks for the review. `NativeWindow` trait might be redundant because `gfx_backend_metal` is actually macOS/iOS specific. And providing native interface gives better extensibility and room for abstraction in future. 👍

File: src/backend/metal/src/lib.rs [hunk 2]
```diff
@@ -84,6 +86,11 @@ impl Instance {
             }))
         }
     }
+
+    pub fn create_surface(&self, window: &winit::Window) -> Surface {
```
Comment (msiglreith) [inline src/backend/metal/src/lib.rs]: Should be feature gated imo.

Comment (davll) [reply src/backend/metal/src/lib.rs]: Oops, my mistake :(

Comment (msiglreith) [reply src/backend/metal/src/lib.rs]: np, thanks for the quick fix!

**Comment by msiglreith** [comment]
Btw do you want to include further changes in this PR (asking because of the `[WIP]`)?
Else the PR looks good to me in the current state and we can proceed merging imo.

**Comment by davll** [comment]
Yes, the PR is done and waiting to be merged :D

**Pull request merged**